docs/high/dataset.rst

```diff
@@ -96,14 +96,70 @@ To retrieve the contents of a `scalar` dataset, you can use the same
 syntax as in NumPy:  ``result = dset[()]``.  In other words, index into
 the dataset using an empty tuple.
 
-For simple slicing, broadcasting is supported:
+For simple slicing, broadcasting is supported::
 
     >>> dset[0,:,:] = np.arange(10)  # Broadcasts to (10,10)
 
 Broadcasting is implemented using repeated hyperslab selections, and is
 safe to use with very large target selections.  It is supported for the above
 "simple" (integer, slice and ellipsis) slicing only.
 
+Compound datasets
+~~~~~~~~~~~~~~~~~
+
+In some cases it may desirable to stored mixed types of data in a single dataset.  Compound
+datasets can be used for this purpose, provided that all of the data in a single column
+is homogenous.  This practice can be valuable when used in conjunction with attributes
+and mapping to generate linked datsets simulating a database structure.
+
+As with simple datasets, new compound datasets are created using either Group.create_dataset()
+or Group.require_dataset().  However, in compound datasets, dtype and shape assignment are not
+optional, and as stated above it is necessary to separate field names from the slices.
+
+Initializing the compound dataset follows the form::
+
+    >>> ds = fn.require_dataset(ds_path, data=HDFData, dtype=DataType, shape=rShape)
```

Comment:
For examples, let's stick to `.create_dataset()` - or the mapping style shorthand `f['name'] = x` - rather than `.require_dataset()`, which is an extra thing to understand.